app/assets/javascripts/discourse/app/mixins/quote-button.js

```diff
@@ -0,0 +1,296 @@
+    this.set("visible", quoteState.buffer.length > 0);
+
+    if (this.canFastEdit) {
+      const quoteRegExp = new RegExp(regexSafeStr(quoteState.buffer), "gi");
+      this._fastEdit(
+        quoteState,
+        quoteRegExp,
+        markdownBody,
+        requiredDataForQuote
+      );
+    }
+
+    // avoid hard loops in quote selection unconditionally
+    // this can happen if you triple click text in firefox
+    if (this._prevSelection === _selectedText) {
+      return;
+    }
+
+    this._prevSelection = _selectedText;
+
+    // on Desktop, shows the button at the beginning of the selection
+    // on Mobile, shows the button at the end of the selection
+    const isMobileDevice = this.site.isMobileDevice;
+    const { isIOS, isAndroid, isSafari, isOpera } = this.capabilities;
+    const showAtEnd = isMobileDevice || isIOS || isAndroid || isOpera;
+
+    // Don't mess with the original range as it results in weird behaviours
+    // where certain browsers will deselect the selection
+    const clone = firstRange.cloneRange();
+
+    // create a marker element containing a single invisible character
+    const markerElement = document.createElement("span");
+    markerElement.appendChild(document.createTextNode("\ufeff"));
+
+    // on mobile, collapse the range at the end of the selection
+    if (showAtEnd) {
+      clone.collapse();
+    }
+    // insert the marker
+    clone.insertNode(markerElement);
+
+    // retrieve the position of the marker
+    const $markerElement = $(markerElement);
+    const markerOffset = $markerElement.offset();
+    const parentScrollLeft = $markerElement.parent().scrollLeft();
+    const $quoteButton = $(this.element);
+
+    // remove the marker
+    const parent = markerElement.parentNode;
+    parent.removeChild(markerElement);
+    // merge back all text nodes so they don't get messed up
+    parent.normalize();
+
+    // work around Safari that would sometimes lose the selection
+    if (isSafari) {
+      this._reselected = true;
+      selection.removeAllRanges();
+      selection.addRange(clone);
+    }
+
+    // change the position of the button
+    schedule("afterRender", () => {
+      if (!this.element || this.isDestroying || this.isDestroyed) {
+        return;
+      }
+
+      let top = markerOffset.top;
+      let left = markerOffset.left + Math.max(0, parentScrollLeft);
+      if (showAtEnd) {
+        top = top + 25;
+        left = Math.min(
+          left + 10,
+          window.innerWidth - this.element.clientWidth - 10
+        );
+      } else {
+        top = top - $quoteButton.outerHeight() - 5;
+      }
+
+      $quoteButton.offset({ top, left });
+    });
+  },
+
+  didInsertElement() {
+    this._super(...arguments);
+
+    const { isWinphone, isAndroid } = this.capabilities;
+    const wait = isWinphone || isAndroid ? INPUT_DELAY : 25;
+    const onSelectionChanged = () => {
+      discourseDebounce(this, this._selectionChanged, wait);
+    };
+
+    $(document)
```

Comment:
Would be good to remove jQuery from the event hooks here as well.

Reply:
Yep will do